### DISCUSSION TITLE
Set environment variables for wasm target?

### DISCUSSION THREAD

**Discussion by derrick56007** [bevyengine/bevy#5737] [Q&A] [answered]
Hello, I'm currently compiling my game to run in the browser, however some variables are read mid-game from env::var(). 

I was wonder how I would go about setting environment variables in the browser

**Comment by wgxh-cli** [comment]
Try to use C/S architecture?

**Reply by derrick56007** [reply]
That is the method I'm trying now. However I'm unsure how I can pass variables to the game from the browser. 

Each url should have a unique id which would pull the respective assets from a server for the game. I'm unsure how to read this unique id from the browser for the wasm.js game to pull the correct assets.

thanks for the help

**Comment by mockersf** [comment, marked as answer]
You won't have access to environment variables from WASM running in the browser.

How do you intend to set those variables for your players? Some javascript variables? Something in the URL fragment? A query to get those variables? A cookie?

You'll have to decide how to expose those variables to your game, and it depends on where they come from.
